lib/interface_quda.cpp

```diff
@@ -3641,26 +3587,24 @@ void invertMultiShiftQuda(void **_hp_x, void *_hp_b, QudaInvertParam *param)
   // Create the solution fields filled with zero
   cudaParam.create = QUDA_ZERO_FIELD_CREATE;
 
-  // now check if we need to invalidate the solutionResident vectors
+  // now check if we need to invalidate the solution vectors
   bool invalidate = false;
-  for (auto v : solutionResident) {
-    if (cudaParam.Precision() != v->Precision()) {
+  for (auto &v : solutionResident) {
+    if (cudaParam.Precision() != v.Precision()) {
       invalidate = true;
       break;
     }
   }
+  if (invalidate) solutionResident.clear();
 
-  if (invalidate) {
-    for (auto v : solutionResident) delete v;
-    solutionResident.clear();
-  }
+  // grow/shrink resident solutions to be correct size
+  size_t old_size = solutionResident.size();
+  solutionResident.resize(param->num_offset);
+  for (int i = old_size; i < param->num_offset; i++) solutionResident[i] = ColorSpinorField(cudaParam);
```

Comment:
std::fill ?

Reply:
I'll apply a better solution: `solutionResident.resize(param->num_offset, cudaParam)`

Reply:
Works for me :-)

Reply:
5f52743d6bc072f87eaaebee86289a5ba740f7cf